src/lib/migrations/lock.ts

```diff
@@ -0,0 +1,42 @@
+import { collections } from "$lib/server/database";
+
+export async function acquireLock(key = "migrations") {
```

Comment:
shouldn't the `key` be the id of the migration ? or it doesn't matter as long as all the functions use same `key` ?

Reply:
I guess the reasoning is to run the migrations in order, and for that one process only should have the lock on _all_ migrations

Reply:
Yes that was the idea! if we need locks for other things in the future we can specify a different key. but migrations should be run in order so one lock for all of them made sense I think